nsis/src/nsis/NsisLoader.java

```diff
@@ -63,15 +64,18 @@ public String getName() {
 									// file
 		List<LoadSpec> loadSpecs = new ArrayList<>();
 		NsisExecutable ne;
-		ne = NsisExecutable.createNsisExecutable(
-				RethrowContinuesFactory.INSTANCE, provider, SectionLayout.FILE);
-
-		if (ne.getHeaderOffset() != -1) {
+		try {
+			ne = NsisExecutable.createNsisExecutable(
+					RethrowContinuesFactory.INSTANCE, provider,
+					SectionLayout.FILE);
 			LoadSpec my_spec = new LoadSpec(this, 0x400000,
 					new LanguageCompilerSpecPair("Nsis:LE:32:default",
 							"default"),
 					true);
 			loadSpecs.add(my_spec);
+		} catch (InvalidFormatException e) {
+			// Not a Nsis file, no loading spec added
```

Comment:
Should an error message be displayed? Otherwise how can user understand that the format is invalid if nothing happens?

(feel free to do that in a follow-up CL, but please leave a TODO comment in this case)